priority_queue.py

```diff
@@ -0,0 +1,50 @@
+from binheap import BinHeap
+from doubly_linked_list import DoublyLinkedList
+
+
+class PriorityQueue:
+    def __init__(self, iterable=None):
+        self._storage = BinHeap()
+        self.priority_values = {}
```

Comment:
priority_values should also be protected.
```suggestion
        self._storage = BinHeap()
        self._priority_values = {}
```